app/com/arpnetworking/metrics/portal/scheduling/impl/OneOffSchedule.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2018 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.scheduling.impl;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import models.internal.scheduling.BaseSchedule;
+import net.sf.oval.constraint.NotNull;
+import net.sf.oval.constraint.ValidateWithMethod;
+
+import java.time.ZonedDateTime;
+import java.util.Optional;
+
+/**
+ * Schedule that should be executed exactly once.
+ *
+ * @author Spencer Pearson (spencerpearson at dropbox dot com)
+ */
+public final class OneOffSchedule extends BaseSchedule {
+
+    private final ZonedDateTime _whenRun;
```

Comment:
I agree that they could use the same field; I separated them because it just felt more consistent with how for `PeriodicSchedule` we had a separate `_offset` field even though we could derive that from `_runAtAndAfter`. But I don't feel strongly, so I'll combine these.

Reply:
This is definitely open for discussion. My question really was how are their definitions different? 

I can see keeping the field and hiding it at the user level. We could even default the field in the builder to that in its parent unless set explicitly. All valid options.

The root question is whether the meaning of the field is actually different in this class than the parent. Granted, your current implementation offers more flexibility, but it's worth asking.